tools/pytorch2onnx.py

```diff
@@ -1,125 +1,144 @@
-            torch.onnx.export(
-                model,
-                inputs,
-                f,
-                operator_export_type=OperatorExportTypes.ONNX_ATEN_FALLBACK,
-                # verbose=True,  # NOTE: uncomment this for debugging
-                # export_params=True,
-            )
-            onnx_model = onnx.load_from_string(f.getvalue())
-
-    # Apply ONNX's Optimization
-    if passes is not None:
-        all_passes = optimizer.get_available_passes()
-        assert all(p in all_passes for p in passes), \
-            f'Only {all_passes} are supported'
-    onnx_model = optimizer.optimize(onnx_model, passes)
-    return onnx_model
+
+
+def pytorch2onnx(model,
+                 input_img,
+                 input_shape,
+                 opset_version=11,
+                 show=False,
+                 output_file='tmp.onnx',
+                 verify=False):
+    model.cpu().eval()
+    # use dummy input to execute model for tracing
+    one_img = mmcv.imread(input_img, 'color')
+    one_img = mmcv.imresize(one_img, input_shape[2:]).transpose(2, 0, 1)
+    # normalize the input images
+    one_img = torch.from_numpy((one_img - 128) / 256).unsqueeze(0).float()
+    (_, C, H, W) = input_shape
+    one_meta = {
+        'img_shape': (H, W, C),
+        'ori_shape': (H, W, C),
+        'pad_shape': (H, W, C),
+        'filename': '<demo>.png',
+        'scale_factor': 1.0,
+        'flip': False
+    }
+    # onnx.export does not support kwargs
+    origin_forward = model.forward
+    model.forward = partial(
+        model.forward, img_metas=[[one_meta]], return_loss=False)
+    # pytorch has some bug in pytorch1.3, we have to fix it
+    # by replacing these existing op
+    register_extra_symbolics(opset_version)
+    torch.onnx.export(
+        model, ([one_img]),
+        output_file,
+        export_params=True,
+        keep_initializers_as_inputs=True,
+        verbose=show,
+        opset_version=opset_version)
+    model.forward = origin_forward
+    print(f'Successfully exported ONNX model: {output_file}')
+    if verify:
+        # check by onnx
+        onnx_model = onnx.load(output_file)
+        onnx.checker.check_model(onnx_model)
+
+        # check the numerical value
+        # get pytorch output
+        pytorch_result = model([one_img], [[one_meta]], return_loss=False)
+
+        # get onnx output
+        input_all = [node.name for node in onnx_model.graph.input]
+        input_initializer = [
+            node.name for node in onnx_model.graph.initializer
+        ]
+        net_feed_input = list(set(input_all) - set(input_initializer))
+        assert (len(net_feed_input) == 1)
+        sess = rt.InferenceSession(output_file)
+        from mmdet.core import bbox2result
+        det_bboxes, det_labels = sess.run(
+            None, {net_feed_input[0]: one_img.detach().numpy()})
+        # only compare a part of result
+        bbox_results = bbox2result(det_bboxes, det_labels, 1)
```

Comment:
We may directly replace `bbox2result` with `[bboxes[labels == i, :] for i in range(num_classes)]` so that we don't need to modify core.

Reply:
> We may directly replace `bbox2result` with `[bboxes[labels == i, :] for i in range(num_classes)]` so that we don't need to modify core.

I think updating the docstring in bbox2result is more directly, as users may not know this is the same logic as bbox2result.